### ISSUE TITLE
Weird issue with annotations Grails 2.5.x + v 2.0.0

### ISSUE THREAD

**Issue by painfulcranium** [grails-plugins/grails-spring-security-core#449]
So I was working with Spring Security Core and Spring REST Security (Using JWT) and annotating controllers in my application (API) with the @Secured annotation with roles I had created.

I started having issues in controllers where I had a user with multiple roles, example ROLE_MANAGER and ROLE_OWNER.

If I annotated the controller with @Secured(['ROLE_OWNER']) but the UserRole object in the DB was in the order such as ROLE_OWNER was id 1 and ROLE_MANAGER was id 2, I was getting a 401 response back. If I swapped the order in the DB so now ROLE_MANAGER is id 1 and ROLE_OWNER is id 2, the access controls seem to work as expected with the annotation, even if I switch the annotation on the controller to be @Secured(['ROLE_MANAGER']) at this point.

Not sure why this is happening, or if there is a weird order thing going on, or if I am only supposed to use 1 role per user (when not using groups too).

Any help would be appreciated as I am trying to figure this out and am not having any luck yet.

I am not using GSP controlled access as the application is an API back-end that is accessed via REST.

**Comment by burtbeckwith** [comment]
If you can create a small test app that demonstrates the problem and upload it to your GitHub (or other online) repo, or attach here (minus the target dir, classes, etc.) I'll take a look

**Comment by painfulcranium** [comment]
OK cool..I will work on a sample app in the morning and get it uploaded..I am curious as to the root of this as it is just strange.

Thanks in advance!

**Comment by painfulcranium** [comment]
Hey Burt, 

Sorry for the noise, was totally my bad. I wrote a check in my code to make sure the roles passed in with a JWT token match what is configured in the DB for a user (in case they get changed mid-flight). The problem was I used an equals() instead of a containsAll() call so the order played a factor between the security context, token, and what was pulled from the DB and in what order.

I will close this out.
